generic/tests/unattended_install.py

```diff
@@ -11,4 +27,57 @@ def run(test, params, env):
     :param params: Dictionary with the test parameters.
     :param env: Dictionary with test environment.
     """
-    unattended_install.run(test, params, env)
+    ret = ThreadError()
+    # use vm name to differentiate among single/multivm
+    vms = params.get("vms").split()
+    nr_vms = len(vms)
+    multivm = nr_vms > 1
+    trigger_time = int(params.get("install_trigger_time", 0))
+    random_trigger = params.get("random_trigger", "no") == "yes"
+    install_timeout = int(params.get("install_timeout", 200))
+    thread_timeout = (1 + 0.05 * nr_vms) * install_timeout
+    ret_lock = threading.RLock()
+    install_threads = []
+
+    def thread_func(vm):
+        """
+        Thread Method to trigger the unattended installation
+
+        :param vm: VM name
+        """
+        try:
+            vm_params = params.object_params(vm)
+            vm_params["main_vm"] = vm
+            unattended_install.run(test, vm_params, env)
+        except Exception as info:
+            logging.error(info)
+            ret_lock.acquire()
+            ret.error()
+            ret_lock.release()
+
+    if multivm:
```

Comment:
Suggest the structure to be:
```python
if not multivm:
    unattended_install.run(test, params, env)
    return

# multivm routine
if not params.get("master_images_clone"):
    test.cancel("provide the param `master_images_clone` to clone"

for vm in vms:
    ...
...
```
because:
1. for single vm variant, it is better to let `unattended_install.run()` raise error directly.
2. to make the code to be updated easily for developers.

Reply:
will make the change.